spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/adapter/AbstractAdaptableMessageListener.java

```diff
@@ -303,13 +302,8 @@ protected void handleResult(InvocationResult resultArg, Message request, Channel
 	 */
 	protected void handleResult(InvocationResult resultArg, Message request, Channel channel, Object source) {
 		if (channel != null) {
-			if (resultArg.getReturnValue() instanceof ListenableFuture) {
-				((ListenableFuture<?>) resultArg.getReturnValue()).addCallback(
-						r -> asyncSuccess(resultArg, request, channel, source, r),
-						t -> asyncFailure(request, channel, t));
-			}
-			else if (resultArg.getReturnValue() instanceof Mono) {
-				((Mono<?>) resultArg.getReturnValue()).subscribe(
+			if (resultArg.getReturnValue() instanceof AsyncReply) {
```

Comment:
Yeah... Sorry for confusing you. I thought about something like inner static class a-la `MonoResultHandler` with two method:
`isMono(resultArg.getReturnValue())` and `handleMono(resultArg.getReturnValue())`.
Of course we need to add `static boolean isReactorPresent` in advance do not load that inner class for potential `NoClassDefFoundError`.

Your solution with extra type and wrappers are good: some potential room for extension, but I believe end-users are going to argue that Framework should care about wrapping by itself. As it is done in the WebFlux for row `Publisher` return type and wrapping into the `Flux`.

We can revert to the required Reactor dependency though if you prefer.